jobs/transforms/user_audits.py

```diff
@@ -39,3 +45,24 @@ def __init__(self):
         # self, like compiled re patterns, ARE LOST AFTER DESERIALIZATION!
         # https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled
         self._pattern = feconf.USER_ID_REGEX
+
+
+@audit_decorators.AuditsExisting(user_models.UserQueryModel)
+class ValidateOldModelsMarkedDeleted(beam.DoFn):
+    """DoFn to validate old models and mark them for deletion"""
+
+    def process(self, input_model):
+        """Function that checks if a model is old enough to mark them deleted.
+
+        Args:
+            input_model: user_models.UserQueryModel. Entity to validate.
+
+        Yields:
+            ModelExpiringError. An error class for expiring models.
+        """
+        model = job_utils.clone_model(input_model)
+        expiration_date = (
+            datetime.datetime.utcnow() -
+            feconf.PERIOD_TO_MARK_MODELS_AS_DELETED)
+        if model.last_updated < expiration_date:
```

Comment:
nit: I think it'd read better if we put expiration date first:
```suggestion
        if expiration_date > model.last_updated:
```

Reply:
Yeah, that makes sense. Changing it

Reply:
Done